val.py

```diff
@@ -330,6 +330,8 @@ def run(
     seen = 0
     confusion_matrix = ConfusionMatrix(nc=nc)
     names = model.names if hasattr(model, "names") else model.module.names  # get class names
+    if single_cls:
+        names = {0: "item"}
```

Comment:
Nice! Does the trick. But... maybe we can catch the places where `names` is used and have a condition there?

The validation class loss seems _off_ because of this change:
<img width="910" alt="image" src="https://github.com/user-attachments/assets/627686ee-1fcd-4cb9-8905-5a6ee6a2ed26">

Reply:
i think this is cause by the fact that the targets class now comes as "0" from the dataloader. 
the output of the model still has classes. 

the compute loss does not break because it is built to make the target cls tensor dim (number of classes), the same shape as the predicted dim.

So we would need to either: 
- create a dataloader just for targets (this solution increases resource demmands and code writing)
- turn off cls_loss if we are dealing with single cls validation.

Turning off cls loss is hard because the compute_loss method accesses the number of classes via the model.
Thtough the targets there is now real mechanism to get this either because the classes come in to the compute_loss methos simply in float form (not one-hot vectors)

Overall i dont think there is a way to make this work with minimal changes. Quick way to fix would be to log cls loss as 0 if we are doing single_cls_val.

Reply:
@kevinconka do you think we log cls loss as 0 to avoid confusions?